benchmark/framework/func_details.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Helper module for details about function being tested."""
+import inspect
+from inspect import Parameter
+from typing import Any, Callable, List
+
+
+class FuncDetails:
+    """Class to introspect some callable details."""
+
+    def __init__(self, func: Callable):
+        """Create an instance."""
+        self.func = func
+
+    def check(self) -> None:
+        """Check for docstring and arguments have default values set."""
+        if not self.doc:
+            raise ValueError("Function docstring is missing")
+
+        if self._arguments[0].name != "control":
```

Comment:
can define "control" as constant on top of file to avoid magic values here.

Reply:
You separate _arguments into two sets, _arguments[0] and _arguments[1:], it might be worth declaring private properties for them and giving them explicit names to make the separation explicit.

Reply:
done:
split on 2 classes:
 one is for basic function details
 another for benchmark: checks for function definition, magic argument name is a constant class variable.